syntax = "proto3";
package sports_keypoints_proto;

import "common.proto";

service OpenCVAndPoseService {
    rpc GetOpenPoseImage(GetOpenPoseImageRequest) returns (GetOpenPoseImageResponse) {}
    rpc GetOpenPoseData(GetOpenPoseDataRequest) returns (GetOpenPoseDataResponse) {}
    rpc GetOpenPoseHandImage(GetOpenPoseHandImageRequest) returns (GetOpenPoseHandImageResponse) {}
    rpc GetOpenPoseHandData(GetOpenPoseHandDataRequest) returns (GetOpenPoseHandDataResponse) {}
    rpc GetOpenPoseAll(GetOpenPoseAllRequest) returns (GetOpenPoseAllResponse) {}

    rpc GetOpenPoseImagesFromVideo(stream GetOpenPoseImageRequest) returns (stream GetOpenPoseImageResponse) {}
    rpc GetOpenPoseDataFromVideo(stream GetOpenPoseDataRequest) returns (stream GetOpenPoseDataResponse) {}
    rpc GetOpenPoseHandImagesFromVideo(stream GetOpenPoseHandImageRequest) returns (stream GetOpenPoseHandImageResponse) {}
    rpc GetOpenPoseHandDataFromVideo(stream GetOpenPoseHandDataRequest) returns (stream GetOpenPoseHandDataResponse) {}
    rpc GetOpenPoseAllFromVideo(stream GetOpenPoseAllRequest) returns (stream GetOpenPoseAllResponse) {}
}

message GetOpenPoseImageRequest {
    bytes image = 1;
}

message GetOpenPoseImageResponse {
    bool success = 1;
    bytes image = 2;
}

message GetOpenPoseDataRequest {
    bytes image = 1;
}

message GetOpenPoseDataResponse {
    bool success = 1;
    Body25PoseKeypoints keypoints = 2;
}

message GetOpenPoseHandImageRequest {
    bytes image = 1;
}

message GetOpenPoseHandImageResponse {
    bool success = 1;
    bytes image = 2;
}

message GetOpenPoseHandDataRequest {
    bytes image = 1;
}

message GetOpenPoseHandDataResponse {
    bool success = 1;
    Body25HandKeypoints keypoints = 2;
}

message GetOpenPoseAllRequest {
    bytes image = 1;
}

message GetOpenPoseAllResponse {
    bool success = 1;
    bytes image = 2;
    Body25PoseKeypoints pose_keypoints = 3;
    Body25HandKeypoints hand_keypoints = 4;
}
